vault/src/issue.rs

```diff
@@ -103,12 +103,41 @@ pub async fn add_keys_from_past_issue_request<B: BitcoinCoreApi + Clone + Send +
     // privkey
     let btc_end_height = bitcoin_core.get_block_count().await? as usize - 1;
 
-    tracing::info!("Rescanning bitcoin chain from height {}...", btc_start_height);
-    for (range_start, range_end) in chunks(btc_start_height, btc_end_height) {
+    // check if the blockchain was pruned after the point where we would need to scan
+    // if it was, we only rescan from the pruned height
+    let btc_pruned_start_height = bitcoin_core.get_pruned_height().await? as usize;
+    let rescan_start_height = btc_start_height.max(btc_pruned_start_height);
+
+    // in parallel, rescan what blockchain we do have stored locally
+    tracing::info!("Rescanning bitcoin chain from height {}...", rescan_start_height);
+    for (range_start, range_end) in chunks(rescan_start_height, btc_end_height) {
```

Comment:
I thought it best to minimise the amount of network calls and maximise local usage. If we remove the scanning altogether, this will affect _all_ vaults and make them query electrs for all their issue requests - is that something we want to do?

Reply:
I think it should still be possible for the Vault to be operated *without* electrs to reduce our dependency on that service.